docs/en/observability/ci-cd-observability.asciidoc

```diff
@@ -0,0 +1,399 @@
+
+[source,bash]
+----
+<project>
+  ...
+  <build>
+    <extensions>
+      <extension>
+          <groupId>io.opentelemetry.contrib</groupId>
+          <artifactId>opentelemetry-maven-extension</artifactId>
+          <version>1.7.0-alpha</version>
+      </extension>
+    </extensions>
+  </build>
+</project>
+----
+
+You can now trigger to send the Maven build reporting performance data to Elastic Observability by
+passing the configuration details as environment variables:
+
+[source,bash]
+----
+export OTEL_EXPORTER_OTLP_ENDPOINT="https://elastic-apm-server.example.com:8200"
+export OTEL_EXPORTER_OTLP_HEADERS="Authorization=Bearer an_apm_secret_token"
+
+mvn verify
+----
+
+You can instrument Maven builds without modifying the pom.xml file using the Maven
+command line argument “-Dmaven.ext.class.path=...”
+
+[source,bash]
+----
+export OTEL_EXPORTER_OTLP_ENDPOINT="https://elastic-apm-server.example.com:8200"
+export OTEL_EXPORTER_OTLP_HEADERS="Authorization=Bearer an_apm_secret_token"
+
+mvn -Dmaven.ext.class.path=path/to/opentelemetry-maven-extension.jar verify
+----
+
+You can also trigger your Maven builds from the CI platform and visualize the end-to-end
+pipeline execution in Elastic Observability, including the detailed steps of your CI
+pipeline and the Maven build.
+
+When invoking Maven builds with Jenkins, it’s unnecessary to use environment variables
+to configure the Maven build (`OTEL_EXPORTER_OTLP_ENDPOINT…`) to rely on the Jenkins capability
+to inject OpenTelemetry configuration as environment variables. For more details, refer to <<ci-cd-install-jenkins>>.
+
+[role="screenshot"]
+.A Jenkins pipeline executing Maven builds
+image::images/jenkins-maven-pipeline.png[Maven builds in Jenkins]
+
+To learn more, see the https://github.com/open-telemetry/opentelemetry-java-contrib/tree/main/maven-extension[integration of Maven builds with Elastic Observability].
+
+[discrete]
+[[ci-cd-ansible]]
+=== Ansible
+
+The Ansible OpenTelemetry plugin integration provides visibility into all your
+Ansible playbooks. The plugin generates traces for each run and performance metrics to help
+you understand which Ansible tasks or roles are run the most, how often they
+fail, and how long they take to complete.
+
+You can configure your Ansible playbooks with the https://docs.ansible.com/ansible/latest/collections/community/general/opentelemetry_callback.html[Ansible OpenTelemetry callback plugin].
+It’s required to install the Opentelemetry python libraries and configure the callback
+as stated in the https://docs.ansible.com/ansible/latest/collections/community/general/opentelemetry_callback.html#examples[examples] section.
+
+The context propagation from the Jenkins job or pipeline is passed to the Ansible run. Therefore,
+everything that happens in the CI is also shown in the traces.
+
+[role="screenshot"]
+.Visibility into your Ansible playbooks
+image::images/jenkins-ansible-pipeline.png[Ansible playbooks in Jenkins]
+
```

Comment:
Do you think we should add an entry regarding the service map for the services used by the ansible playbooks?

[This PR ](https://github.com/ansible-collections/community.general/pull/3558/) already supports it.

For instance, the ansible playbooks executed in one of Jenkins controllers produced this service map

![image](https://user-images.githubusercontent.com/2871786/143047967-aac6deaf-48f8-4431-8126-3639f458b291.png)


Thats something we get for free with the ansible opentelemetry callback plugin

Reply:
@cyrille-leclerc What do you think?^^

Reply:
it's a great idea @v1v thanks!

Reply:
See https://github.com/elastic/observability-docs/pull/1266/commits/7c1be8f15a08d0771cc221f88554feb54152c42d